modules/apps/configuration-admin/configuration-admin-web/src/main/java/com/liferay/configuration/admin/web/internal/search/ConfigurationModelIndexer.java

```diff
@@ -371,16 +369,7 @@ protected void doReindex(String[] ids) throws Exception {
 			_configurationModelRetriever.getConfigurationModels(
 				ExtendedObjectClassDefinition.Scope.SYSTEM, null);
 
-		Set<Document> documents = new HashSet<>();
-
-		for (ConfigurationModel configurationModel :
-				configurationModels.values()) {
-
-			documents.add(getDocument(configurationModel));
-		}
-
-		_indexWriterHelper.updateDocuments(
-			CompanyConstants.SYSTEM, documents, false);
+		reindex(configurationModels.values());
```

Comment:
ya i thought about doing this, the early returns kind of made me hesitate, didnt necessarily want to change the behavior too much. this is probably the right way to do it, i just wasnt able to look deep enough. in general this framework needs some attention, so we have to draw a line somewhere so we dont get carried away with the changes